src/neurotechdevkit/results/_results.py

```diff
@@ -420,6 +424,182 @@ def render_steady_state_amplitudes_3d(
+        """Validate that slicing axis and position are within scenario range.
+
+        `slice_axis` should be either 0, 1, or 2 (for X, Y, Z).
+        `slice_position` must be within boundaries for `slice_axis` extent.
+
+        Args:
+            slice_axis: the axis along which to slice the 3D field to be recorded. If
+                None, then the complete field will be recorded. Use 0 for X axis, 1
+                for Y axis and 2 for Z axis.
+            slice_position: the position (in meters) along the slice axis at
+                which the slice of the 3D field should be made.
+
+        Raises:
+            ValueError if axis is not 0, 1, 2.
+            ValueError if `slice_position` falls outside the current range of
+                `slice_axis`.
+            ValueError if  `slice_axis` is None but `slice_position` is not None and
+                vice versa.
+        """
+        if slice_axis not in (0, 1, 2):
+            raise ValueError(
+                "Unexpected value received for `slice_axis`. ",
+                "Expected axis are 0 (X), 1 (Y) and/or 2 (Z).",
+            )
+        if (slice_axis is None and slice_position is not None) or (
+            slice_axis is not None and slice_position is None
+        ):
+            raise ValueError(
+                "Both `slice_axis` and `slice_position` must be passed together "
+                "to correctly define how to slice the field. "
+            )
+
+        origin = np.array(self.scenario.origin, dtype=float)
+        extent = self.scenario.extent
+
+        current_range = (origin[slice_axis], origin[slice_axis] + extent[slice_axis])
+        if (slice_position < current_range[0]) or (slice_position > current_range[1]):
+            raise ValueError(
+                "`slice_position` is out of range for `slice_axis`. ",
+                f"Received value {slice_position} and "
+                f"current range is {current_range}.",
+            )
+
+    def get_steady_state_result_2d(
+        self,
+        slice_axis: Optional[SliceAxis] = None,
+        slice_position: Optional[float] = None,
+    ) -> SteadyStateResult2D:
+        """Return a 2D steady-state result from a 3D steady-state result.
+
+        This function will take a slice of the 3D steady-state result and return a 2D
+        steady-state result of the slice. A slice through the scenario can be specified
+        via `slice_axis` and `slice_position`. Eg. to take a slice at z=0.01 m, use
+        `slice_axis=2` and `slice_position=0.01`. If `slice_axis` and `slice_position`
+        are not specified, the slice will be taken along the scenario's `slice_axis` and
+        `slice_position`.
+
+        !!! warning
+            When taking a 2D slice of a 3D steady-state result, the sources and target
+            are not being accounted for and will not be preserved in the 2D result.
+
+        Args:
+            slice_axis (Optional[SliceAxis], optional): The axis along which to slice
```

Comment:
nit (minor comment):
```suggestion
            slice_axis (Optional[SliceAxis]): The axis along which to slice
```

To go further, it's not 100% necessary to put the type in the docstring if you have the type-hint already (but NDK docstring convention does do so, so it also makes sense to keep it).